x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/async/StoredAsyncTask.java

```diff
@@ -95,4 +95,15 @@ protected synchronized void onFailure(Exception e) {
     public void cancelTask(TaskManager taskManager, Runnable runnable, String reason) {
         taskManager.cancelTaskAndDescendants(this, reason, true, ActionListener.wrap(runnable));
     }
+
+    public static QlStatusResponse getStatusResponse(StoredAsyncTask<?> asyncTask) {
```

Comment:
I would introduce it one level above so we can keep this task generic and usable by ML.